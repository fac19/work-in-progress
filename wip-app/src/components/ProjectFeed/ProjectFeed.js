import React, { useEffect } from "react";
import ProjectCard from "../../components/ProjectCard/ProjectCard";
import { ProjectGrid } from "./ProjectFeed.style";

const ProjectFeed = ({ projects }) => {
  const renderProjects = (projects) => {
    return projects.map(({ project_name, project_status, id }) => {
      return (
        <ProjectCard
          project_name={project_name}
          project_status={project_status}
          key={id}
        />
      );
    });
  };

  return (
    <>
      <ProjectGrid>{renderProjects(projects)}</ProjectGrid>
    </>
  );
};

export default ProjectFeed;
