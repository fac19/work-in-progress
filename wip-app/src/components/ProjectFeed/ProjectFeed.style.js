import styled from "styled-components";

const ProjectGrid = styled.section`
  margin: 0 5%;
  display: grid;
  justify-content: center;
  grid-template-columns: repeat(auto-fit, minmax(15em, 30em));
  grid-gap: 32px;
`;

export { ProjectGrid };
