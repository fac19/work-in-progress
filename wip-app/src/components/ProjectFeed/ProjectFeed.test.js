import React from "react";
import TestRenderer from "react-test-renderer";
import ProjectFeed from "./ProjectFeed";

describe("Snapshot test", () => {
  const projects = [{ id: 1 }, { id: 2 }];
  it("renders the correct ui", () => {
    const component = TestRenderer.create(<ProjectFeed projects={projects} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
