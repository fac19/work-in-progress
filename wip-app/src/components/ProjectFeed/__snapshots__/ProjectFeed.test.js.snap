// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<section
  className="sc-AxhUy fejNsb"
>
  <article
    className="sc-AxjAm gMVhDx"
    onClick={[Function]}
  >
    <img
      alt="undefined's project: undefined"
      className="sc-AxiKw bGHWDv"
    />
    <h3
      className="sc-AxirZ lcIoiq"
    />
    <p
      className="sc-AxhCb goFehh"
    >
      By 
    </p>
  </article>
  <article
    className="sc-AxjAm gMVhDx"
    onClick={[Function]}
  >
    <img
      alt="undefined's project: undefined"
      className="sc-AxiKw bGHWDv"
    />
    <h3
      className="sc-AxirZ lcIoiq"
    />
    <p
      className="sc-AxhCb goFehh"
    >
      By 
    </p>
  </article>
</section>
`;
